### PR TITLE
Update CI and and related documentation

### PR DESCRIPTION
**Pull request by alerque**

### CODE REVIEW

**Comment by alerque** [comment]
I'm going to (at least temporarily) throw in the towel on setting up a GH Action for this. In particular that shouldn't hold up the next release. I have gotten it to work several ways that work for various situations, but each way has major cons. The containerized way I used for Luarocks does not work well here because Busted doesn't just need it's *own* dependencies but all the dependencies of every project it has to run on. That means dependency injection into the container and that's a mess.

I'll break out the related work out of this PR and we can look into other implementation methods that might be more generally useful to end users.

File: .github/workflows/busted.yml
Comment (Tieske) [inline .github/workflows/busted.yml:10]:
```suggestion
# different busted version.
```

File: .github/workflows/busted.yml
Comment (Tieske) [inline .github/workflows/busted.yml:24]: any reason to comment out the openresty version?

File: .github/workflows/busted.yml
Comment (Tieske) [inline .github/workflows/busted.yml:48]: I think the lib-ev, copas, luasec, lua-ev dependencies are based on Busted 1.x which had support for async testing. I think they can go (not 100% sure)

File: .github/workflows/busted.yml
Comment (Tieske) [inline .github/workflows/busted.yml:55]:
```suggestion
        # disable the paths to force use of installed rockspec
        run: busted -c -v --lpath="" --cpath="" --Xoutput "--color"'
```

Comment (Tieske) [reply .github/workflows/busted.yml:55]: the coverage option in `.busted` should be removed

File: busted-scm-2.rockspec
Comment (Tieske) [inline busted-scm-2.rockspec:42]: curently also luacov, unless it is removed from `.busted`

**Review by Tieske — COMMENTED** [review]
Looks good, some small comments.

File: .github/workflows/busted.yml
Comment (alerque) [reply .github/workflows/busted.yml:24]: Yes. Something is wrong with the @leafo's openresty image. It regularly but seemingly randomly crashes (maybe once every ten runs on small projects up to almost 100% on some specific large ones). There is some segfault problem that doesn't happen with openresty locally only in the GH runner images. Many other projects have temporarily disabled this too. We had to do so recently for luasocket because it was crashing there and blocking merges. Disabling it here is preemptive but I think also reasonable given the number of false positive test failures it generates elsewhere.

Comment (Tieske) [reply .github/workflows/busted.yml:24]: ok, that makes sense then

File: .github/workflows/busted.yml
Comment (alerque) [reply .github/workflows/busted.yml:55]: This project doesn't even have a `.busted`. Per explanation in [this comment](https://github.com/lunarmodules/say/pull/16#discussion_r952379886) I'm not in favor of adding one either but of course if it did the coverage option should not be enabled.

File: busted-scm-2.rockspec
Comment (alerque) [reply busted-scm-2.rockspec:42]: c.f. https://github.com/lunarmodules/busted/pull/689#discussion_r952411360

File: .github/workflows/busted.yml
Comment (alerque) [reply .github/workflows/busted.yml:48]: Hmm, okay I'll look into that.

Comment (Tieske) [reply .github/workflows/busted.yml:48]: can do this later

**Review by Tieske — APPROVED** [review]

**Comment by Tieske** [comment]
The merge after approval now leaves the CI in a failed state. Some build jobs should probably not run? @alerque

**Comment by alerque** [comment]
I'll look into it.

**Pull request merged**
